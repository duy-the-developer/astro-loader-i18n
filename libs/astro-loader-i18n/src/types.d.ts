export type NormalizeLocales<T extends Locales> = {
  [K in keyof T]: T[K] extends string ? T[K] : T[K] extends {
      codes: Array<string>;
  } ? T[K]['codes'][number] : never;
}[number];
