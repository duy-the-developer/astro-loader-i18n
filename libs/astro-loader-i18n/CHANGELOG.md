# astro-loader-i18n

## 0.2.5

### Patch Changes

- f8c74ef: Simplify parseDataProxy

## 0.2.4

### Patch Changes

- bd303c5: Add source maps

## 0.2.3

### Patch Changes

- bc5c8ea: Enhance error messages
- 38134d0: Improve translation id generation

## 0.2.2

### Patch Changes

- 21d4dad: Enhance type packaging

## 0.2.1

### Patch Changes

- 6056aaa: Export extendI18nLoaderSchema as localizedSchema
- b0cc973: Upgrade dependencies

## 0.2.0

### Minor Changes

- aa75291: Enhance README.md
- e406dbb: Improve `i18nPropsAndParams` type-safety

### Patch Changes

- bc18996: Add getStaticPaths helpers

## 0.1.0

### Minor Changes

- b745ea0: Add i18n loader and schemas

## 0.0.5

### Patch Changes

- c9ffa98: Distribute LICENSE and README.md

## 0.0.4

### Patch Changes

- dc7abbd: Add release management scripts

## 0.0.3

### Patch Changes

- 708b192: Fix release management

## 0.0.2

### Patch Changes

- 3c5f233: Add release management
