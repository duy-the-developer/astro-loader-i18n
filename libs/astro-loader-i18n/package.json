{
  "name": "astro-loader-i18n",
  "version": "0.2.5",
  "description": "An Astro content loader for i18n files and folder structures.",
  "keywords": [
    "astro",
    "astro-content-loader",
    "astro-loader",
    "i18n",
    "withastro"
  ],
  "homepage": "https://github.com/openscript/astro-loader-i18n",
  "license": "MIT",
  "author": "Robin BÃ¼hler",
  "repository": "github:openscript/astro-loader-i18n",
  "type": "module",
  "main": "./dist/astro-loader-i18n.js",
  "types": "./dist/astro-loader-i18n.d.ts",
  "exports": {
    ".": {
      "import": "./dist/astro-loader-i18n.js",
      "types": "./dist/astro-loader-i18n.d.ts"
    }
  },
  "files": [
    "dist",
    "README.md",
    "LICENSE"
  ],
  "peerDependencies": {
    "astro": "^5.5.0",
    "limax": "^4.1.0"
  },
  "devDependencies": {
    "@codecov/vite-plugin": "^1.9.0",
    "@eslint/js": "^9.25.1",
    "@types/node": "^22.15.3",
    "@vitest/coverage-v8": "^3.1.2",
    "astro": "^5.7.9",
    "eslint": "^9.25.1",
    "globals": "^16.0.0",
    "limax": "^4.1.0",
    "tslib": "^2.8.1",
    "typescript": "^5.8.3",
    "typescript-eslint": "^8.31.1",
    "vite": "^6.3.4",
    "vite-plugin-dts": "^4.5.3",
    "vitest": "^3.1.2"
  },
  "publishConfig": {
    "access": "public"
  }
}
