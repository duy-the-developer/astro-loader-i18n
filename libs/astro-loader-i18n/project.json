{
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "name": "astro-loader-i18n",
  "sourceRoot": "libs/astro-loader-i18n/src",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@nx/vite:build",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "libs/astro-loader-i18n/dist"
      }
    },
    "dev": {
      "executor": "@nx/vite:build",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "libs/astro-loader-i18n/dist",
        "watch": true
      }
    },
    "prepare": {
      "dependsOn": ["build"],
      "command": "cp README.md LICENSE libs/astro-loader-i18n"
    },
    "test": {
      "executor": "@nx/vite:test",
      "outputs": ["{options.reportsDirectory}"],
      "options": {
        "reportsDirectory": "./coverage",
        "coverage": true
      }
    },
    "lint": {
      "executor": "@nx/eslint:lint",
      "outputs": ["{options.outputFile}"]
    }
  },
  "tags": []
}
