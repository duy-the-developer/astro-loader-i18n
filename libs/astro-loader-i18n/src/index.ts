export { i18nLoader } from "./loaders/i18n-loader";
export { i18nInfileSchema } from "./schemas/i18n-infile-schema";
export { i18nLoaderSchema } from "./schemas/i18n-loader-schema";
export { localePaths } from "./paths/locale-paths";
export { translationPaths } from "./paths/translation-paths";
export { mergeGetStaticPathsResults } from "./utils/merge";
export { createCollection } from "./collections/create-collection";
