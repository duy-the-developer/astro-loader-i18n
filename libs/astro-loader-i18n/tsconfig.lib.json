{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../../dist/out-tsc",
    "declaration": true,
    "types": ["node", "vite/client"]
  },
  "include": ["src/**/*.ts"],
  "exclude": ["vite.config.ts", "test"]
}
