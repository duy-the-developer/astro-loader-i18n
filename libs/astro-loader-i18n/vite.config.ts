import { defineConfig } from "vite";
import { nxViteTsPaths } from '@nx/vite/plugins/nx-tsconfig-paths.plugin';
import dts from "vite-plugin-dts";
import path from "path";

export default defineConfig({
  root: __dirname,
  cacheDir: '../../node_modules/.vite/libs/astro-loader-i18n',
  plugins: [nxViteTsPaths(), dts({ entryRoot: "src", tsconfigPath: path.join(__dirname, "tsconfig.lib.json") })],
  build: {
    outDir: '../../dist/libs/astro-loader-i18n',
    emptyOutDir: true,
    lib: {
      entry: "src/index.ts",
      name: "astro-loader-i18n",
      formats: ["es"],
    },
    rollupOptions: {
      external: ["astro/loaders", "astro/zod"],
      output: {
        globals: {
          "astro/loaders": "astroLoaders",
          "astro/zod": "astroZod",
        },
      }
    }
  },

  test: {
    globals: true,
    environment: 'node',
    coverage: {
      reportsDirectory: '../../coverage/libs/astro-loader-i18n',
    },
    reporters: ['verbose'],
    include: ['test/**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}'],
  },
});
