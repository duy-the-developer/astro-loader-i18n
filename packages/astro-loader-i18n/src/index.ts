export { i18nFolderLoader } from "./loaders/i18n-folder-loader";
export { i18nSchema } from "./schemas/i18n-schema";
