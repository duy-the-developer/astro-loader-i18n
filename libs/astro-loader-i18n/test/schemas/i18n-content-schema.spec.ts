import { expect, describe, it } from "vitest";
import { localized } from "../../src/astro-loader-i18n";
import { z } from "astro/zod";

describe("i18nContentSchema", () => {
  it("should extend the schema", async () => {
    const schema = localized(z.string(), ["en", "fr", "de"] as const);
    expect(JSON.stringify(schema)).toMatchSnapshot();
    expect(schema.options[0].shape).toMatchSnapshot();
  });
  it("should extend the schema and make partial if option is set", async () => {
    const schema = localized(z.string(), ["en", "fr", "de"] as const, true);
    expect(JSON.stringify(schema)).toMatchSnapshot();
    expect(schema.options[0].shape).toMatchSnapshot();
  });
});
