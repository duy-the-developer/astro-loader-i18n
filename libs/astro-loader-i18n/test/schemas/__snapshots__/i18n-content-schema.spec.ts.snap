// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`i18nContentSchema > should extend the schema 1`] = `"{"_def":{"options":[{"_def":{"unknownKeys":"strip","catchall":{"_def":{"typeName":"ZodNever"},"~standard":{"version":1,"vendor":"zod"}},"typeName":"ZodObject"},"~standard":{"version":1,"vendor":"zod"},"_cached":null},{"_def":{"checks":[],"typeName":"ZodString","coerce":false},"~standard":{"version":1,"vendor":"zod"}}],"typeName":"ZodUnion"},"~standard":{"version":1,"vendor":"zod"}}"`;

exports[`i18nContentSchema > should extend the schema 2`] = `
{
  "de": ZodString {
    "_def": {
      "checks": [],
      "coerce": false,
      "typeName": "ZodString",
    },
    "and": [Function],
    "array": [Function],
    "brand": [Function],
    "catch": [Function],
    "default": [Function],
    "describe": [Function],
    "isNullable": [Function],
    "isOptional": [Function],
    "nullable": [Function],
    "nullish": [Function],
    "optional": [Function],
    "or": [Function],
    "parse": [Function],
    "parseAsync": [Function],
    "pipe": [Function],
    "promise": [Function],
    "readonly": [Function],
    "refine": [Function],
    "refinement": [Function],
    "safeParse": [Function],
    "safeParseAsync": [Function],
    "spa": [Function],
    "superRefine": [Function],
    "transform": [Function],
    "~standard": {
      "validate": [Function],
      "vendor": "zod",
      "version": 1,
    },
  },
  "en": ZodString {
    "_def": {
      "checks": [],
      "coerce": false,
      "typeName": "ZodString",
    },
    "and": [Function],
    "array": [Function],
    "brand": [Function],
    "catch": [Function],
    "default": [Function],
    "describe": [Function],
    "isNullable": [Function],
    "isOptional": [Function],
    "nullable": [Function],
    "nullish": [Function],
    "optional": [Function],
    "or": [Function],
    "parse": [Function],
    "parseAsync": [Function],
    "pipe": [Function],
    "promise": [Function],
    "readonly": [Function],
    "refine": [Function],
    "refinement": [Function],
    "safeParse": [Function],
    "safeParseAsync": [Function],
    "spa": [Function],
    "superRefine": [Function],
    "transform": [Function],
    "~standard": {
      "validate": [Function],
      "vendor": "zod",
      "version": 1,
    },
  },
  "fr": ZodString {
    "_def": {
      "checks": [],
      "coerce": false,
      "typeName": "ZodString",
    },
    "and": [Function],
    "array": [Function],
    "brand": [Function],
    "catch": [Function],
    "default": [Function],
    "describe": [Function],
    "isNullable": [Function],
    "isOptional": [Function],
    "nullable": [Function],
    "nullish": [Function],
    "optional": [Function],
    "or": [Function],
    "parse": [Function],
    "parseAsync": [Function],
    "pipe": [Function],
    "promise": [Function],
    "readonly": [Function],
    "refine": [Function],
    "refinement": [Function],
    "safeParse": [Function],
    "safeParseAsync": [Function],
    "spa": [Function],
    "superRefine": [Function],
    "transform": [Function],
    "~standard": {
      "validate": [Function],
      "vendor": "zod",
      "version": 1,
    },
  },
}
`;

exports[`i18nContentSchema > should extend the schema and make partial if option is set 1`] = `"{"_def":{"options":[{"_def":{"unknownKeys":"strip","catchall":{"_def":{"typeName":"ZodNever"},"~standard":{"version":1,"vendor":"zod"}},"typeName":"ZodObject"},"~standard":{"version":1,"vendor":"zod"},"_cached":null},{"_def":{"checks":[],"typeName":"ZodString","coerce":false},"~standard":{"version":1,"vendor":"zod"}}],"typeName":"ZodUnion"},"~standard":{"version":1,"vendor":"zod"}}"`;

exports[`i18nContentSchema > should extend the schema and make partial if option is set 2`] = `
{
  "de": ZodOptional {
    "_def": {
      "description": undefined,
      "errorMap": [Function],
      "innerType": ZodString {
        "_def": {
          "checks": [],
          "coerce": false,
          "typeName": "ZodString",
        },
        "and": [Function],
        "array": [Function],
        "brand": [Function],
        "catch": [Function],
        "default": [Function],
        "describe": [Function],
        "isNullable": [Function],
        "isOptional": [Function],
        "nullable": [Function],
        "nullish": [Function],
        "optional": [Function],
        "or": [Function],
        "parse": [Function],
        "parseAsync": [Function],
        "pipe": [Function],
        "promise": [Function],
        "readonly": [Function],
        "refine": [Function],
        "refinement": [Function],
        "safeParse": [Function],
        "safeParseAsync": [Function],
        "spa": [Function],
        "superRefine": [Function],
        "transform": [Function],
        "~standard": {
          "validate": [Function],
          "vendor": "zod",
          "version": 1,
        },
      },
      "typeName": "ZodOptional",
    },
    "and": [Function],
    "array": [Function],
    "brand": [Function],
    "catch": [Function],
    "default": [Function],
    "describe": [Function],
    "isNullable": [Function],
    "isOptional": [Function],
    "nullable": [Function],
    "nullish": [Function],
    "optional": [Function],
    "or": [Function],
    "parse": [Function],
    "parseAsync": [Function],
    "pipe": [Function],
    "promise": [Function],
    "readonly": [Function],
    "refine": [Function],
    "refinement": [Function],
    "safeParse": [Function],
    "safeParseAsync": [Function],
    "spa": [Function],
    "superRefine": [Function],
    "transform": [Function],
    "~standard": {
      "validate": [Function],
      "vendor": "zod",
      "version": 1,
    },
  },
  "en": ZodOptional {
    "_def": {
      "description": undefined,
      "errorMap": [Function],
      "innerType": ZodString {
        "_def": {
          "checks": [],
          "coerce": false,
          "typeName": "ZodString",
        },
        "and": [Function],
        "array": [Function],
        "brand": [Function],
        "catch": [Function],
        "default": [Function],
        "describe": [Function],
        "isNullable": [Function],
        "isOptional": [Function],
        "nullable": [Function],
        "nullish": [Function],
        "optional": [Function],
        "or": [Function],
        "parse": [Function],
        "parseAsync": [Function],
        "pipe": [Function],
        "promise": [Function],
        "readonly": [Function],
        "refine": [Function],
        "refinement": [Function],
        "safeParse": [Function],
        "safeParseAsync": [Function],
        "spa": [Function],
        "superRefine": [Function],
        "transform": [Function],
        "~standard": {
          "validate": [Function],
          "vendor": "zod",
          "version": 1,
        },
      },
      "typeName": "ZodOptional",
    },
    "and": [Function],
    "array": [Function],
    "brand": [Function],
    "catch": [Function],
    "default": [Function],
    "describe": [Function],
    "isNullable": [Function],
    "isOptional": [Function],
    "nullable": [Function],
    "nullish": [Function],
    "optional": [Function],
    "or": [Function],
    "parse": [Function],
    "parseAsync": [Function],
    "pipe": [Function],
    "promise": [Function],
    "readonly": [Function],
    "refine": [Function],
    "refinement": [Function],
    "safeParse": [Function],
    "safeParseAsync": [Function],
    "spa": [Function],
    "superRefine": [Function],
    "transform": [Function],
    "~standard": {
      "validate": [Function],
      "vendor": "zod",
      "version": 1,
    },
  },
  "fr": ZodOptional {
    "_def": {
      "description": undefined,
      "errorMap": [Function],
      "innerType": ZodString {
        "_def": {
          "checks": [],
          "coerce": false,
          "typeName": "ZodString",
        },
        "and": [Function],
        "array": [Function],
        "brand": [Function],
        "catch": [Function],
        "default": [Function],
        "describe": [Function],
        "isNullable": [Function],
        "isOptional": [Function],
        "nullable": [Function],
        "nullish": [Function],
        "optional": [Function],
        "or": [Function],
        "parse": [Function],
        "parseAsync": [Function],
        "pipe": [Function],
        "promise": [Function],
        "readonly": [Function],
        "refine": [Function],
        "refinement": [Function],
        "safeParse": [Function],
        "safeParseAsync": [Function],
        "spa": [Function],
        "superRefine": [Function],
        "transform": [Function],
        "~standard": {
          "validate": [Function],
          "vendor": "zod",
          "version": 1,
        },
      },
      "typeName": "ZodOptional",
    },
    "and": [Function],
    "array": [Function],
    "brand": [Function],
    "catch": [Function],
    "default": [Function],
    "describe": [Function],
    "isNullable": [Function],
    "isOptional": [Function],
    "nullable": [Function],
    "nullish": [Function],
    "optional": [Function],
    "or": [Function],
    "parse": [Function],
    "parseAsync": [Function],
    "pipe": [Function],
    "promise": [Function],
    "readonly": [Function],
    "refine": [Function],
    "refinement": [Function],
    "safeParse": [Function],
    "safeParseAsync": [Function],
    "spa": [Function],
    "superRefine": [Function],
    "transform": [Function],
    "~standard": {
      "validate": [Function],
      "vendor": "zod",
      "version": 1,
    },
  },
}
`;
